string/Cursor.ts

```diff
@@ -0,0 +1,61 @@
+/** Describes a string and an index within it. */
+export class Cursor {
+  #input: { value: string };
+
+  constructor(
+    input: string,
+    public index: number = 0,
```

Comment:
What should happen if `index` is out-of-bounds? Right now I allow it to happen without error, but I'm torn. Are there sensible use cases for referencing an index after the end of the string, or before 0?

Reply:
This will be resolved by #97 merging first.